tests/frontend/onnx/test_context.py

```diff
@@ -0,0 +1,189 @@
+import onnx
+import pytest
+
+try:
+    from onnx import (
+        TensorProto,
+        ValueInfoProto,
+        helper,
+    )
+
+    from xdsl.frontend.onnx.context import (
+        OnnxXdslMapping,
+        build_module,
+        visit_graph,
+        visit_node,
+        visit_value_info,
+    )
+except ImportError as exc:
+    print(exc)
+    pytest.skip("onnx is an optional dependency", allow_module_level=True)
+
+
+def test_visit_node_unknown_op_name():
+    """
+    Test for unknown expected onnx op
+    """
+    node_attributes = {
+        "name": "dummy_name",
+        "op_type": "dummy_op",
+    }
+
+    node = helper.make_node(
+        **node_attributes, inputs=["input1", "input2"], outputs=["output"]
+    )
+    ctx = OnnxXdslMapping()
+    with pytest.raises(ValueError, match="Unknown ONNX op name dummy_op"):
+        visit_node(node=node, ctx=ctx)
+
+
+def test_visit_node_add():
+    # initialize context
+    ctx = OnnxXdslMapping()
+
+    # create graph composed only of one Add operation
+    graph, add_node = _create_graph_binary_op("Add", "add_graph")
+
+    # visit graph
+    visit_graph(graph, ctx)
+
+    # visit node (test passes if no exceptions are raised)
+    visit_node(add_node, ctx)
+
+
+def test_visit_node_sub():
+    # initialize context
+    ctx = OnnxXdslMapping()
+
+    # create graph composed only of one Sub operation
+    graph, sub_node = _create_graph_binary_op("Sub", "sub_graph")
+
+    # visit graph
+    visit_graph(graph, ctx)
+
+    # visit node (test passes if no exceptions are raised)
+    visit_node(sub_node, ctx)
```

Comment:
I might be missing something but it seems like the effect of this is not tested. Am I understanding this correctly?

Reply:
Done